src/components/events/event_new/fields/CostField.js

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Form, Col } from 'react-bootstrap';
+import CurrencyInput from 'react-currency-input-field';
+import '../EventNewForm.css';
+
+export class CostField extends Component {
+  constructor(props) {
+    super(props);
+    this.receiveModeOfTransport = this.receiveModeOfTransport.bind(this);
+    this.receiveCost = this.receiveCost.bind(this);
+  }
+
+  /***************************** Core functions *****************************/
+
+  /**
+   * Receives the mode of transportation to use.
+   * @param mode of transportation to use.
+   */
+  receiveModeOfTransport(mode) {
+    this.props.handleChange('method', mode.target.value);
+  }
+
+  /**
+   * Receives the cost of the event.
+   * @param cost of event.
+   */
+  receiveCost(cost) {
+    this.props.handleChange('cost', cost);
+  }
+
+  /**************************** Visual component ****************************/
+
+  render() {
+    if (this.props.eventType === 'tran')
+      // display mode of transportation & cost
+      return (
+        <Form.Group controlId="exampleForm.ControlSelect1">
+          <Form.Row>
+            <Col>
+              <Form.Label>Mode of transportation</Form.Label>
+              <Form.Control as="select" onChange={this.receiveModeOfTransport}>
+                <option disabled>Select your option</option>
+                <option value={'plane'}>Plane</option>
+                <option value={'train'}>Train</option>
+                <option value={'automobile'}>Automobile</option>
+                <option value={'bus'}>Bus</option>
+              </Form.Control>
```

Comment:
It confuses me that the "CostField" object has the transportation mode inside of it. However, I recognize that it probably looks way better on the front end by grouping elements together in this way. Thus, I'd suggest making a comment at the top of the file at least, indicating why you're grouping the two elements together (I'm not opposed to keeping it this way).

Reply:
I can do that. My logic behind nesting these two things together are:
- it's much easier for us to provide the fields in columns this way
- mode of transportation is a controlled field which just complements the cost of a tran event
I'm open to discussion about further abstracting this if anyone has other ideas, but for now I'll add more documentation and leave it be.

Reply:
I am really not a fan... I mean, you are going to be using CostField in places that don't refer to transportation right? This just feels like logic that should be totally separated... There are always ways to make things work on the front-end, so I reallyyyy wouldn't worry about separating these from that side of things...

Reply:
This is true. So, if I separate cost `Mode` into its own component, I'm thinking the form in `EventNewComponent` will have to look something like this...

```
  <Form>
            <ToggleUseDates
              eventType={this.state.eventType}
              toggleUseDates={this.toggleUseDates}
            />
            <DateField
              begin_time={this.state.begin_time}
              end_time={this.state.end_time}
              useDates={this.state.useDates}
              handleChange={this.handleChange}
            />
            <LocationField
              eventType={this.state.eventType}
              handleChange={this.handleChange}
            />
            <CostField
              eventType={this.state.eventType}
              handleChange={this.handleChange}
            />
            <FormGroup>
              <Form.Row>
                <Mode
                  eventType={this.state.eventType}
                  handleChange={this.handleChange}
                />
                <CostField
                  eventType={this.state.eventType}
                  handleChange={this.handleChange}
                />
              </Form.Row>
            </FormGroup>
            <DescriptionField
              eventType={this.state.eventType}
              useDates={this.state.useDates}
              handleChange={this.handleChange}
            />
</Form>
```